tests/broadcast.rs

```diff
@@ -1,164 +1,236 @@
+) where
     A: Adversary<Broadcast<NodeId>>,
-    F: Fn(BTreeMap<NodeId, Arc<NetworkInfo<NodeId>>>) -> A,
+    F: Fn() -> A,
 {
     let mut rng = rand::thread_rng();
     let sizes = (1..6)
         .chain(once(rng.gen_range(6, 20)))
         .chain(once(rng.gen_range(30, 50)));
     for size in sizes {
+        // cloning since it gets moved into a closure
+        let cloned_netinfo_map = adversary_netinfo.clone();
         let num_faulty_nodes = util::max_faulty(size);
-        let num_good_nodes = size - num_faulty_nodes;
         info!(
             "Network size: {} good nodes, {} faulty nodes",
-            num_good_nodes, num_faulty_nodes
+            size - num_faulty_nodes,
+            num_faulty_nodes
         );
-        let adversary = |adv_nodes| new_adversary(adv_nodes);
-        let network = TestNetwork::new(num_good_nodes, num_faulty_nodes, adversary, new_broadcast);
-        test_broadcast(network, proposed_value);
+
+        let (net, _) = NetBuilder::new(0..size as u16)
+            .num_faulty(num_faulty_nodes as usize)
+            .message_limit(10_000 * size as usize)
+            .no_time_limit()
+            .adversary(new_adversary())
+            .using(move |info| {
+                let netinfo = Arc::new(info.netinfo);
+                cloned_netinfo_map
+                    .lock()
+                    .unwrap()
+                    .insert(info.id, netinfo.clone());
+                Broadcast::new(netinfo, 0).expect("Failed to create a ThresholdSign instance.")
+            })
+            .build(&mut rng)
+            .expect("Could not construct test network.");
+
+        test_broadcast(net, proposed_value, &mut rng);
     }
 }
 
 #[test]
 fn test_8_broadcast_equal_leaves_silent() {
-    let adversary = |_| SilentAdversary::new(MessageScheduler::Random);
+    let new_adversary = || ReorderingAdversary::new();
+    let mut rng = rand::thread_rng();
+    let size = 8;
+    let (net, _) = NetBuilder::new(0..size as u16)
+        .num_faulty(0 as usize)
+        .message_limit(10_000 * size as usize)
+        .no_time_limit()
+        .adversary(new_adversary())
+        .using(move |node_info: NewNodeInfo<_>| {
+            Broadcast::new(Arc::new(node_info.netinfo), 0)
+                .expect("Failed to create a ThresholdSign instance.")
+        })
+        .build(&mut rng)
+        .expect("Could not construct test network.");
+
     // Space is ASCII character 32. So 32 spaces will create shards that are all equal, even if the
     // length of the value is inserted.
-    test_broadcast(
-        TestNetwork::new(8, 0, adversary, new_broadcast),
-        &[b' '; 32],
-    );
+    test_broadcast(net, &[b' '; 32], &mut rng);
 }
 
 #[test]
 fn test_broadcast_random_delivery_silent() {
-    let new_adversary = |_| SilentAdversary::new(MessageScheduler::Random);
-    test_broadcast_different_sizes(new_adversary, b"Foo");
+    let new_adversary = || ReorderingAdversary::new();
```

Comment:
That closure (and maybe also the variable) seems unnecessary.
```rust
let new_adversary = ReorderingAdversary::new;
```

Reply:
Eliminated